src/Operation/RemoteMergeOperation.php

```diff
@@ -171,7 +190,31 @@ private function createSourceBranch()
         }
 
         if ($this->squash) {
-            $this->gitHelper->squashCommits($this->targetBase, $sourceBranch, $this->forceSquash);
+            $this->gitHelper->squashCommits($this->targetBase, $sourceBranch, $this->forceSquash, $this->guardSync);
+        }
+
+        $currentBaseHeadCommit = $this->processHelper->runCommand(['git', 'rev-parse', $this->targetBase]);
+        $lastKnownCommonCommit = $this->processHelper->runCommand(['git', 'merge-base', '--fork-point', $this->targetBase, $sourceBranch]);
+
+        if ($currentBaseHeadCommit !== $lastKnownCommonCommit) {
+            if ($this->rebase) {
+                try {
+                    $this->processHelper->runCommand(['git', 'pull', '--rebase', $this->targetBase]);
+                } catch (\Exception $e) {
+                    // Error, abort the rebase operation
+                    $this->processHelper->runCommand(['git', 'rebase', '--abort'], true);
+
+                    throw new MergeWorkflowException(sprintf('Git rebase failed while trying to synchronize history against "%s".', $this->targetBase), 0, $e);
+                }
+
+                // Retrieve the commits again
+                $currentBaseHeadCommit = $this->processHelper->runCommand(['git', 'rev-parse', $this->targetBase]);
+                $lastKnownCommonCommit = $this->processHelper->runCommand(['git', 'merge-base', '--fork-point', $this->targetBase, $sourceBranch]);
```

Comment:
This is unnecessary to get the commits again, since if rebase is enabled, guardSync will be disabled, which means this is unnecessary processing for values that won't be used.